packages/map/src/marker/hooks.js

```diff
@@ -0,0 +1,15 @@
+import { useContext, useEffect } from 'react';
+import MarkerContext from './context';
+
+
+export const useMarkerContext = () => useContext(MarkerContext);
+
+export const useMarkerEffect = (fn, deps = []) => {
+  const marker = useMarkerContext();
+  deps.push(marker);
```

Comment:
is there any reason why you are mutating the argument? Would prefer `[marker, ...deps]`